### PR TITLE
Kill the light client, CHTs and change tries.

### PR DESCRIPTION
**Pull request by arkpar**
This PR removes:
* Light client leftovers
* CHTs. These are not used currently. Smoldot uses warp sync proofs instead.
* Change tries. Not used by any existing chains. Will probably require a new implementation anyway.

polkadot companion: https://github.com/paritytech/polkadot/pull/4191

### CODE REVIEW

**Comment by gilescope** [comment]
This should be unblocked now. #4105 has landed and now the diff looks a fair bit smaller.

File: primitives/io/src/lib.rs [hunk 1]
```diff
@@ -195,16 +195,9 @@ pub trait Storage {
...
-	///
-	/// The hashing algorithm is defined by the `Block`.
-	///
-	/// Returns `Some(Vec<u8>)` which holds the SCALE encoded hash or `None` when
-	/// changes trie is disabled.
-	fn changes_root(&mut self, parent_hash: &[u8]) -> Option<Vec<u8>> {
-		self.storage_changes_root(parent_hash)
-			.expect("Invalid `parent_hash` given to `changes_root`.")
+	/// Always returns `None`. This function exists for compatibility reasons.
+	fn changes_root(&mut self, _parent_hash: &[u8]) -> Option<Vec<u8>> {
```
Comment (arkpar) [inline primitives/io/src/lib.rs:199]: @bkchr Is there a way to remove this without breaking compatibility with existing chains? E.g. when loading polkadot genesis wasm, we check that the host function must exist, even though it is never called.

File: primitives/state-machine/src/overlayed_changes/changeset.rs [hunk 2]
```diff
@@ -69,7 +69,6 @@ struct InnerValue<V> {
 	/// Current value. None if value has been deleted.
 	value: V,
 	/// The set of extrinsic indices where the values has been changed.
-	/// Is filled only if runtime has announced changes trie support.
 	extrinsics: Extrinsics,
```
Comment (cheme) [inline primitives/state-machine/src/overlayed_changes/changeset.rs:72]: Should we just remove this `Extrinsics` field (I remember thinking of some optimization to it, but not having it is a better optimization).

Comment (arkpar) [reply primitives/state-machine/src/overlayed_changes/changeset.rs:72]: I thinks this is also involved in supporting transactions in the runtime. I.e. to revert storage changes when extrinsic fails. In any case I'd rather do complex logic changes and optimisations in a separate PR.

Comment (cheme) [reply primitives/state-machine/src/overlayed_changes/changeset.rs:72]: +1 for separate, IIRC transaction stuff just merge extrinsic when commiting (it is just additional computation).

**Comment by dvdplm** [comment]
@andresilva @bkchr @sorpaas @adoerr Would be fantastic to see this land soon, anything we can do to help clear up the remaining doubts? :)

**Comment by arkpar** [comment]
@joao-paulo-parity Not sure why the companion check fails here with `Failed to detect our crate "sc-light" referenced in polkadot`. `sc-light` crate has been removed from substrate in this PR, and the companion PR also removes its usage in polkadot.

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
@arkpar I think we can ignore. Just let us merge this and then update the companion using the old school way ;)

**Review by cheme — APPROVED** [review]

**Pull request merged**